ui/pages/confirmations/components/simulation-details/useBalanceChanges.ts

```diff
@@ -124,7 +124,9 @@ function getTokenBalanceChanges(
     };
 
     const decimals =
-      asset.standard === TokenStandard.ERC20 ? erc20Decimals[asset.address] : 0;
+      asset.standard === TokenStandard.ERC20 && erc20Decimals[asset.address]
```

Comment:
Should we go deeper and ask why there is missing decimals for an address?

When we query the decimals for a single token in `fetchErc20Decimals`, we fallback to `ERC20_DEFAULT_DECIMALS` if the response is bad, or if there is an error, so I'm not sure how this could happen.

We also wouldn't want to fallback to zero decimals if the token was an ERC-20, only for any other standard.

Reply:
@jpuri, echoing @matthewwalsh0's point, previously we set decimals to zero only for non-ERC20 tokens, like ERC721 or ERC1155. This is because ERC721 and ERC1155 tokens are indivisible and cannot have decimals. This change, however, introduces fallback logic that sets the decimals to zero for some ERC20s, which isn't our goal. 

Matt also mentioned that it's unexpected for erc20Decimals[asset.address] to be undefined, as we're supposed to fetch decimals for all ERC20 addresses through fetchErc20Decimals, where we already have a default handling mechanism.

It would be better to look into why this is happening rather than adding more fallback logic.